### DISCUSSION TITLE
Do anybody use oil nvim with lazyvim?

### DISCUSSION THREAD

**Discussion by Demianeen** [LazyVim/LazyVim#2822] [Q&A] [answered]
I heard recently about oil nvim and want to try, but I feel like it would be another deep hole of my time for a few days😂. Does somebody already have oil nvim configured with lazyvim?

**Comment by jyuan0** [comment]
There's a PR: https://github.com/LazyVim/LazyVim/pull/2702

**Reply by Demianeen** [reply]
Yeah, but I don't really see any integration there. It is just initializes oil.nvim and add key for it
![CleanShot 2024-03-26 at 11 24 19@2x](https://github.com/LazyVim/LazyVim/assets/51330172/8bd15168-5da0-4130-ab20-40c727ae1ece)

I mean we have a whole lazyvim ecosystem with cmd and root dir opening, lsp actions and all of the other stuff. I hoped to test oil with at least some of those features, more like a replacement for neo-tree

**Reply by surmish** [reply]
There is nothing committed in that PR. Was it closed as intended?

**Comment by quangdn42** [comment, marked as answer]
I would suggest trying out mini.files, it also allows file navigation with a buffer similar to oil, unless you're looking for something more specific.

Mini.files is already available as an extra.

**Comment by alewis001** [comment]
I've started using oil recently...

```
return {
  "stevearc/oil.nvim",
  keys = {
    { "<leader>e", "<cmd>Oil --float<CR>", desc = "Explorer" },
  },
  opts = {
    view_options = {
      show_hidden = true,
    },
    float = {
      padding = 5,
    },
  },
  -- Optional dependencies
  dependencies = { "nvim-tree/nvim-web-devicons" },
}
```
I did also disable neo-tree so I reused the keymap for oil; the default I think is more typically `"-"`.

I also prefer the floating window but you can easily drop the `--float` from above if you want the more typical usage.

**Reply by Demianeen** [reply]
But how do you deal with lsp renaming in oil? Does it support it by default?

**Reply by alewis001** [reply]
😄 Good question. I must admit I haven't tried it so I don't know what the answer is sorry.

**Reply by ernstwi** [reply]
Should work by default! Neovim's builtin LSP client has `vim.lsp.buf.rename()`, which I'm guessing is what oil uses. This API works regardless of how LSP is initialised, so no special integration with LazyVim's LSP stuff should be necessary.

**Comment by Old-Farmer** [comment]
I think oil.nvim is an alternative to netrw. Neo-tree is another kind of plugin which is more powerful to manage your fs. I prefer treat oil.nvim as an enhancement and leave neo-tree in your config.

**Comment by erlangparasu** [comment]
How to move files or folders using `oil.nvim` ?
In `nvim-tree` just simple by using `x` and `p`

**Reply by Demianeen** [reply]
you just yank or delete them as normal text in the buffer and oil should figure it out on its own

**Comment by Kurren123** [comment]
Any success in using this?

**Reply by Demianeen** [reply]
Yes. I think everything works out of the box

**Reply by Kurren123** [reply]
@Demianeen did you need to disable netRW?

**Reply by Demianeen** [reply]
Not, sure. I use lazyvim, and I just enabled oil.nvim

**Reply by lvl4ul2i** [reply]
You can have both, no need to disable netrw. The one thing you have to consider is if you want to have oil as a default file explorer to edit directories. From the oil docs:
```lua
  -- Oil will take over directory buffers (e.g. `vim .` or `:e src/`)
  -- Set to false if you want some other plugin (e.g. netrw) to open when you edit directories.
  default_file_explorer = true,
```

**Reply by surmish** [reply]
What I've been using:

```lua
  {
    "stevearc/oil.nvim",
    lazy = false,
    dependencies = { "nvim-tree/nvim-web-devicons" },
    keys = {
      { "<leader>fe", "<cmd>Oil<CR>", desc = "File Explorer" },
    },
    opts = {
      default_file_explorer = true,
      view_options = {
        show_hidden = true,
      },
    },
  },
```

**Comment by rasibn** [comment]
If someone else also wants to use both Oil and neo-tree together but use a neo-tree keybind like `<leader>e` for Oil, you might need to disable the keybind for neo-tree. Overriding in Oil opts is not enough.

```lua
  {
    "nvim-neo-tree/neo-tree.nvim",
    keys = {
      { "<leader>e", false },
    },
  },
```

I think LazyVim loads neo-tree with autocmd which causes a race condition between it's default keybind and whatever you override it with. 
https://github.com/LazyVim/LazyVim/blob/13a4a84e3485a36e64055365665a45dc82b6bf71/lua/lazyvim/plugins/editor.lua#L43
